scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -1467,6 +1467,24 @@ def test_align_vectors_parallel():
     assert_allclose(R.apply(b[0]), a[0], atol=atol)
 
 
+def test_align_vectors_antiparallel():
+    # Test exact 180 deg rotation
+    atol = 1e-12
+    a = [[0, 1, 0], [-1, 0, 0]]
+    b = [[0, -1, 0], [-1, 0, 0]]
+    R, _ = Rotation.align_vectors(a, b, weights=[np.inf, 1])
+    assert np.isclose(R.magnitude(), np.pi, atol=atol)
```

Comment:
Just a surface-level comment for now, but could you use the built-in NumPy testing functions for making these kinds of assertions? The feedback is much better, even for scalars:

```python
>>> import numpy as np
>>> from numpy.testing import assert_allclose
>>> assert np.isclose(3.1, np.pi)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
>>> assert_allclose(3.1, np.pi)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/numpy/testing/_private/utils.py", line 1684, in assert_allclose
    assert_array_compare(compare, actual, desired, err_msg=str(err_msg),
  File "/home/treddy/github_projects/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.3.0/python-3.11.2-4qncg2nqev5evxfmamde3e6rnb34b4ls/lib/python3.11/contextlib.py", line 81, in inner
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/numpy/testing/_private/utils.py", line 885, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 1 / 1 (100%)
Max absolute difference among violations: 0.04159265
Max relative difference among violations: 0.01323935
 ACTUAL: array(3.1)
 DESIRED: array(3.141593)
```